plugins/osdn/node_firewall.go

```diff
@@ -0,0 +1,102 @@
+package osdn
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+
+	utildbus "k8s.io/kubernetes/pkg/util/dbus"
+	kexec "k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/iptables"
+	utilwait "k8s.io/kubernetes/pkg/util/wait"
+)
+
+type FirewallRule struct {
+	table string
+	chain string
+	args  []string
+}
+
+type Firewall struct {
```

Comment:
"firewall" isn't the right name (for the struct or the file); half of the iptables rules are to *un*-firewall traffic, and the other half are NAT rules that are needed regardless of firewalling. So the name should be something with "iptables", not "firewall"